sdk/core/core-http/src/serializer.ts

```diff
@@ -1231,37 +1348,92 @@ export interface BaseMapper {
   constraints?: MapperConstraints;
 }
 
+/**
+ * Mappers are definitions of the Data models used in the library.
+ * These data models are part of the Operation or Client definitions in the responses or parameters.
+ */
 export type Mapper = BaseMapper | CompositeMapper | SequenceMapper | DictionaryMapper | EnumMapper;
 
+/**
+ * Helps identify how to transform from one set of properties in the source object, to a set of properties in the target object.
+ */
 export interface PolymorphicDiscriminator {
+  /**
+   * Name used during serialization.
+   */
   serializedName: string;
+  /**
+   * Name of the client.
```

Comment:
this is the "friendly name" of the discriminant, since the json property name might be something ugly like `@odata.kind` and we want to just say `kind` on the final object